observation_portal/requestgroups/viewsets.py

```diff
@@ -183,36 +187,60 @@ def max_allowable_ipp(self, request):
 
     @action(detail=False, methods=['post'])
     def cadence(self, request):
+        request_serializer = self.get_request_serializer(data=request.data, context={'request': request})
```

Comment:
I might be reading this wrong, but checking that this serializer is valid before expanding a cadence I think would mean that every request that gets passed in would have to contain a `cadence`? So that `else` on line 202 will never be reached

Reply:
Yeah that looks right now. I think we used to think this could take a RequestGroup with multiple requests, only one of which had a cadence block in it to get expanded, but I don't think that ever really made sense. It seems like we should just limit the input to a single request with a cadence block for this expansion endpoint

Reply:
That seems reasonable to me. Our API docs even say it's supposed to take only a single request https://developers.lco.global/#generate-cadence so hopefully that means that everyone who is using the endpoint is only submitting it with a single Request? Impossible to say for sure though